File: mobile-web/src/api/debox-account-friend.ts
/*用户好友管理 */

import request from '@/utils/request';
import { SERVER } from '@/constants/http';
import {
  CreateDeboxAccountFriendReq,
  CreateDeboxAccountFriendResp,
  DeleteDeboxAccountFriendReq,
  DeleteDeboxAccountFriendResp,
  GetDeboxAccountFriendReq,
  GetDeboxAccountFriendResp,
  GetDeboxAccountFriendsReq,
  GetDeboxAccountFriendsResp,
  SendPrivateMessageTextReq,
  SendPrivateMessageTextResp,
  SyncDeboxAccountFriendsReq,
  SyncDeboxAccountFriendsResp,
  UpdateDeboxAccountFriendReq,
  UpdateDeboxAccountFriendResp,
  UpdateDeboxAccountFriendStatusReq,
  UpdateDeboxAccountFriendStatusResp,
} from '@/typings/debox-account-friend';

export const DeboxAccountFriendApi = {
  // 获取账号好友列表
  list: async (data: GetDeboxAccountFriendsReq): Promise<GetDeboxAccountFriendsResp> => {
    const response = await request({
      url: `${SERVER}/debox/debox-account-friends/list`,
      method: 'POST',
      data,
    });
    return response.data;
  },

  // 获取账号好友信息
  info: async (data: GetDeboxAccountFriendReq): Promise<GetDeboxAccountFriendResp> => {
    const response = await request({
      url: `${SERVER}/debox/debox-account-friends/${data.id}`,
      method: 'GET',
      params: {},
    });
    return response.data;
  },

  // 创建账号好友
  create: async (data: CreateDeboxAccountFriendReq): Promise<CreateDeboxAccountFriendResp> => {
    const response = await request({
      url: `${SERVER}/debox/debox-account-friends`,
      method: 'POST',
      data,
    });
    return response.data;
  },

  // 更新账号好友信息
  update: async (data: UpdateDeboxAccountFriendReq): Promise<UpdateDeboxAccountFriendResp> => {
    const response = await request({
      url: `${SERVER}/debox/debox-account-friends/update`,
      method: 'PUT',
      data,
    });
    return response.data;
  },

  // 更新账号好友状态
  updateStatus: async (data: UpdateDeboxAccountFriendStatusReq): Promise<UpdateDeboxAccountFriendStatusResp> => {
    const response = await request({
      url: `${SERVER}/debox/debox-account-friends/update-status`,
      method: 'PUT',
      data,
    });
    return response.data;
  },

  // 删除账号好友
  delete: async (data: DeleteDeboxAccountFriendReq): Promise<DeleteDeboxAccountFriendResp> => {
    const response = await request({
      url: `${SERVER}/debox/debox-account-friends/${data.id}`,
      method: 'DELETE',
      data: {},
    });
    return response.data;
  },

  // 同步好友列表
  syncFriends: async (data: SyncDeboxAccountFriendsReq): Promise<SyncDeboxAccountFriendsResp> => {
    const response = await request({
      url: `${SERVER}/debox/debox-account-friends/sync-friends`,
      method: 'POST',
      data,
    });
    return response.data;
  },

  // 发送私聊文本消息
  sendPrivateMessageText: async (data: SendPrivateMessageTextReq): Promise<SendPrivateMessageTextResp> => {
    const response = await request({
      url: `${SERVER}/debox/debox-account-friends/send-private-message-text`,
      method: 'POST',
      data,
    });
    return response.data;
  },
};

File: apps/crates/service_hub/debox/src/controller/debox_account_friend.rs
//! DeBox账号好友管理
use axum_context::Context;

use axum_response::{Responder, Response};
use axum_validator::{Extension, Json, Path};
use inject::AInjectProvider;

use crate::{
    DeboxAccountFriendService,
    dto::debox_account_friend::{
        CreateDeboxAccountFriendReq, CreateDeboxAccountFriendResp, DeleteDeboxAccountFriendReq,
        DeleteDeboxAccountFriendResp, GetDeboxAccountFriendReq, GetDeboxAccountFriendResp,
        GetDeboxAccountFriendsReq, GetDeboxAccountFriendsResp, SyncDeboxAccountFriendsReq,
        SyncDeboxAccountFriendsResp, UpdateDeboxAccountFriendReq, UpdateDeboxAccountFriendResp,
        UpdateDeboxAccountFriendStatusReq, UpdateDeboxAccountFriendStatusResp,
    },
};

/// 控制器
pub struct DeboxAccountFriendController;

impl DeboxAccountFriendController {
    /// 获DeBox账号好友列表
    pub async fn list(
        ctx: Context,
        Extension(provider): Extension<AInjectProvider>,
        Json(req): Json<GetDeboxAccountFriendsReq>,
    ) -> Responder<GetDeboxAccountFriendsResp> {
        let debox_account_friend_service: DeboxAccountFriendService = provider.provide();
        let (results, total) = debox_account_friend_service.list(&ctx, req).await?;

        let resp = Response::data((results, total).into());
        Ok(resp)
    }

    /// 获取DeBox账号好友信息
    pub async fn info(
        ctx: Context,
        Extension(provider): Extension<AInjectProvider>,
        Path(req): Path<GetDeboxAccountFriendReq>,
    ) -> Responder<GetDeboxAccountFriendResp> {
        let debox_account_friend_service: DeboxAccountFriendService = provider.provide();
        let result = debox_account_friend_service.info(&ctx, req).await?;

        let resp = Response::data(result.into());
        Ok(resp)
    }

    /// 添加DeBox账号好友
    pub async fn create(
        ctx: Context,
        Extension(provider): Extension<AInjectProvider>,
        Json(req): Json<CreateDeboxAccountFriendReq>,
    ) -> Responder<CreateDeboxAccountFriendResp> {
        let debox_account_friend_service: DeboxAccountFriendService = provider.provide();
        let _result = debox_account_friend_service.create(&ctx, req).await?;

        let resp = Response::ok();
        Ok(resp)
    }

    /// 更新DeBox账号好友
    pub async fn update(
        ctx: Context,
        Extension(provider): Extension<AInjectProvider>,
        Json(req): Json<UpdateDeboxAccountFriendReq>,
    ) -> Responder<UpdateDeboxAccountFriendResp> {
        let debox_account_friend_service: DeboxAccountFriendService = provider.provide();
        let _result = debox_account_friend_service.update(&ctx, req).await?;

        let resp = Response::ok();
        Ok(resp)
    }

    /// 更新DeBox账号好友状态
    pub async fn update_status(
        ctx: Context,
        Extension(provider): Extension<AInjectProvider>,
        Json(req): Json<UpdateDeboxAccountFriendStatusReq>,
    ) -> Responder<UpdateDeboxAccountFriendStatusResp> {
        let debox_account_friend_service: DeboxAccountFriendService = provider.provide();
        debox_account_friend_service
            .update_status(&ctx, req)
            .await?;

        let resp = Response::ok();
        Ok(resp)
    }

    /// 删除DeBox账号好友
    pub async fn delete(
        ctx: Context,
        Extension(provider): Extension<AInjectProvider>,
        Path(req): Path<DeleteDeboxAccountFriendReq>,
    ) -> Responder<DeleteDeboxAccountFriendResp> {
        let debox_account_friend_service: DeboxAccountFriendService = provider.provide();
        let _result = debox_account_friend_service.delete(&ctx, req).await?;

        let resp = Response::ok();
        Ok(resp)
    }
}

impl DeboxAccountFriendController {
    /// 同步好友列表
    pub async fn sync_friends(
        ctx: Context,
        Extension(provider): Extension<AInjectProvider>,
        Json(req): Json<SyncDeboxAccountFriendsReq>,
    ) -> Responder<SyncDeboxAccountFriendsResp> {
        let debox_account_friend_service: DeboxAccountFriendService = provider.provide();
        debox_account_friend_service.sync_friends(&ctx, req).await?;

        let resp = Response::ok();
        Ok(resp)
    }
}
